Source/ThirdParty/ANGLE/src/libANGLE/renderer/metal/mtl_command_buffer.mm

```diff
@@ -1132,8 +1132,13 @@ inline void CheckPrimitiveType(MTLPrimitiveType primitiveType)
 #if ANGLE_MTL_EVENT_AVAILABLE
 void CommandBuffer::setEventImpl(id<MTLEvent> event, uint64_t value)
 {
-    ASSERT(!mActiveCommandEncoder);
-    [get() encodeSignalEvent:event value:value];
+    ASSERT(!mActiveCommandEncoder || mActiveCommandEncoder->getType() != CommandEncoder::RENDER);
+    // For non-render command encoder, we can safely end it, so that we can encode a signal
+    // event.
+    forceEndingCurrentEncoder();
+
+    if (event)
```

Comment:
@kkinnunen-apple @djg we are sending a nil event down to `setEventImpl`, so skipping it is probably wrong, but I'm unfamiliar with where this event gets sent from